source/lmp/tests/model_convert.py

```
@@ -0,0 +1,83 @@
+    output.parent.mkdir(parents=True, exist_ok=True)
+    lock_file = output.with_name(f".{output.name}.lock")
+    started = time.monotonic()
+
+    while True:
+        if _is_up_to_date(source, output):
+            return output
+        try:
+            fd = os.open(str(lock_file), os.O_CREAT | os.O_EXCL | os.O_WRONLY)
+        except FileExistsError:
+            if time.monotonic() - started >= _LOCK_TIMEOUT_SECONDS:
+                raise TimeoutError(f"Timed out waiting for {lock_file}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)